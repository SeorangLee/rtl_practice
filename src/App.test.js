import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import Form from './pages/Home/Form';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';


function setupApp(){
  return  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
}

function setupForm(){
  return render(
    <MemoryRouter>
      <Form />
    </MemoryRouter>
  )
}
describe('<App />',()=>{
  // it('<App /> snapshot', () => {
  //   const utils = render(<App />);
  //   expect(utils.container).toMatchSnapshot();
  // });

  it('renders learn react link', () => {
    setupApp();
    // const link = screen.getByRole('link', {name: /how it works/i});
    // userEvent.click(link);
    const subredditInput = screen.getByLabelText('r /');
    userEvent.type(subredditInput, 'reactjs');
    const submitButton = screen.getByRole('button', { name: /search/i });
    userEvent.click(submitButton);
    expect(screen.getByText(/is loading/i)).toBeInTheDocument();
    screen.debug();

  });

})


// describe('<Form />', ()=>{
  
//   it('loads posts and renders them on the page', ()=>{
//     setupForm();


//     screen.debug();
//   })
// })