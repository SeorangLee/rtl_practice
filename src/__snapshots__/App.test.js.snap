// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> <App /> snapshot 1`] = `
<div>
  <header
    class="sc-AxjAm hXjNkZ"
  >
    <a
      href="/"
    >
      <svg
        class="sc-AxirZ iCeOHL"
      >
        logo.svg
      </svg>
    </a>
    <nav>
      <a
        class="sc-AxiKw ejNrOs"
        href="/how-it-works"
      >
        How it works
      </a>
      <a
        class="sc-AxiKw ejNrOs"
        href="/about"
      >
        About
      </a>
    </nav>
  </header>
  <main>
    <div
      class="sc-AxhCb iQZeXb"
    >
      <section
        class="sc-AxhUy bmKebj"
      >
        <h1
          class="sc-AxgMl hHcRn"
        >
          Find the top posts on Reddit
        </h1>
        <form
          class="sc-fzozJi gsgKPd"
        >
          <label
            class="sc-fzoLsD gcKKKR"
          >
            r /
            <input
              class="sc-fzpans hAQcze"
              name="subreddit"
              type="text"
              value="javascript"
            />
          </label>
          <button
            class="sc-AxmLO ckZYVg"
            type="submit"
          >
            Search
          </button>
        </form>
      </section>
    </div>
  </main>
</div>
`;

exports[`<App /> renders learn react link 1`] = `
<div>
  <header
    class="sc-AxjAm hXjNkZ"
  >
    <a
      href="/"
    >
      <svg
        class="sc-AxirZ iCeOHL"
      >
        logo.svg
      </svg>
    </a>
    <nav>
      <a
        class="sc-AxiKw ejNrOs"
        href="/how-it-works"
      >
        How it works
      </a>
      <a
        class="sc-AxiKw ejNrOs"
        href="/about"
      >
        About
      </a>
    </nav>
  </header>
  <main>
    <div
      class="sc-AxhCb iQZeXb"
    >
      <section
        class="sc-AxhUy bmKebj"
      >
        <h1
          class="sc-AxgMl hHcRn"
        >
          Find the top posts on Reddit
        </h1>
        <form
          class="sc-fzozJi gsgKPd"
        >
          <label
            class="sc-fzoLsD gcKKKR"
          >
            r /
            <input
              class="sc-fzpans hAQcze"
              name="subreddit"
              type="text"
              value="javascript"
            />
          </label>
          <button
            class="sc-AxmLO ckZYVg"
            type="submit"
          >
            Search
          </button>
        </form>
      </section>
    </div>
  </main>
</div>
`;

exports[`renders learn react link 1`] = `
<div>
  <header
    class="sc-AxjAm hXjNkZ"
  >
    <a
      href="/"
    >
      <svg
        class="sc-AxirZ iCeOHL"
      >
        logo.svg
      </svg>
    </a>
    <nav>
      <a
        class="sc-AxiKw ejNrOs"
        href="/how-it-works"
      >
        How it works
      </a>
      <a
        class="sc-AxiKw ejNrOs"
        href="/about"
      >
        About
      </a>
    </nav>
  </header>
  <main>
    <div
      class="sc-AxhCb iQZeXb"
    >
      <section
        class="sc-AxhUy bmKebj"
      >
        <h1
          class="sc-AxgMl hHcRn"
        >
          Find the top posts on Reddit
        </h1>
        <form
          class="sc-fzozJi gsgKPd"
        >
          <label
            class="sc-fzoLsD gcKKKR"
          >
            r /
            <input
              class="sc-fzpans hAQcze"
              name="subreddit"
              type="text"
              value="javascript"
            />
          </label>
          <button
            class="sc-AxmLO ckZYVg"
            type="submit"
          >
            Search
          </button>
        </form>
      </section>
    </div>
  </main>
</div>
`;
